github/gists.go

```diff
@@ -52,6 +52,30 @@ func (g GistFile) String() string {
 	return Stringify(g)
 }
 
+// GistCommit represents a commit on a gist.
+type GistCommit struct {
+	Version      *string        `json:"version,omitempty"`
+	User         *User          `json:"user,omitempty"`
+	ChangeStatus map[string]int `json:"change_status,omitempty"`
```

Comment:
`map[string]int` is great for dynamic content, where you don't know in advance what fields may or may not be present.

Based on my reading of the Gists API, it seems like change status is always three fields:

- "deletions"
- "additions"
- "total"

Is that not the case?

If so, it's much better to provide a static API for it. So, this line:

```Go
ChangeStatus map[string]int `json:"change_status,omitempty"`
```

Can become:

```Go
ChangeStatus *ChangeStatus `json:"change_status,omitempty"`
```

And you'd add that new struct:

```Go
// ChangeStatus represents the change status of a gist commit.
type ChangeStatus struct {
    Deletions *int `json:"deletions,omitempty"`
    Additions *int `json:"additions,omitempty"`
    Total     *int `json:"total,omitempty"`
}
```

That way, users of this library can look at at endpoint that returns `GistCommit` and immediately know what values will be populated inside `ChangeStatus`. When it's a map, you can't know that as easily. Also, editor auto-completion will help them discover those 3 values without leaving their editor. That's why it's an improvement that's worth doing whenever possible.

**Edit:** Actually, I just spotted a very similar struct [`CommitStats`](https://godoc.org/github.com/google/go-github/github#CommitStats). There is a chance we might want to reuse it instead of creating a separate one for Gist Commits. We should do that if we reasonably expect the two to stay in sync and that they represent the same thing. Even though they have slightly different names.

I'd probably lean towards trying to reuse it for now, and we can always create a dedicated `ChangeStatus` for gist later, should we notice the two diverge. So, this line can be simply replaced with:

```Go
ChangeStatus *ChangeStats `json:"change_status,omitempty"`
```